Classes/Service/NodeService.php

```diff
@@ -0,0 +1,219 @@
+use Neos\ContentRepository\Domain\Repository\WorkspaceRepository;
+use Neos\ContentRepository\Domain\Service\NodeTypeManager;
+use Neos\ContentRepository\Domain\Utility\NodePaths;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Mvc\Controller\ControllerContext;
+use Neos\Neos\Controller\CreateContentContextTrait;
+use NEOSidekick\AiAssistant\Dto\FocusKeywordModuleConfigurationDto;
+use NEOSidekick\AiAssistant\Dto\FocusKeywordModuleResultDto;
+use NEOSidekick\AiAssistant\Dto\ResultCollectionDto;
+use NEOSidekick\AiAssistant\Factory\FocusKeywordModuleResultDtoFactory;
+
+class NodeService
+{
+    use CreateContentContextTrait;
+
+    private const MIXIN_NODE_TYPE = 'NEOSidekick.AiAssistant:Mixin.AiPageBriefing';
+
+    /**
+     * @Flow\Inject
+     * @var EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @Flow\Inject
+     * @var WorkspaceRepository
+     */
+    protected $workspaceRepository;
+
+    /**
+     * @Flow\Inject
+     * @var FocusKeywordModuleResultDtoFactory
+     */
+    protected $focusKeywordModuleResultDtoFactory;
+
+    /**
+     * @Flow\Inject
+     * @var NodeTypeManager
+     */
+    protected $nodeTypeManager;
+
+    public function getNodesThatNeedProcessing(FocusKeywordModuleConfigurationDto $configurationDto, ControllerContext $controllerContext): ResultCollectionDto
+    {
+        $workspace = $this->workspaceRepository->findByIdentifier($configurationDto->getWorkspace());
+
+        if (!$workspace) {
+            // throw
+        }
+
+        $workspaceChain = array_merge([$workspace], array_values($workspace->getBaseWorkspaces()));
+        $queryBuilder = $this->createQueryBuilder($workspaceChain);
+        $queryBuilder->andWhere('n.nodeType IN (:includeNodeTypes)');
+        $queryBuilder->setParameter('includeNodeTypes', $this->getNodeTypeFilter($configurationDto));
+        $iterator = $queryBuilder->getQuery()->iterate();
+
+        $nodeDatasThatNeedProcessing = [];
+        $nodeDatasThatNeedProcessingCount = 0;
+        $iteratedItems = 0;
+
+        /** @var NodeData $nodeData */
+        foreach ($this->iterate($iterator) as $nodeData) {
+            $iteratedItems++;
+            if ($iteratedItems <= $configurationDto->getFirstResult()) {
+                continue;
+            }
+
+            if ($nodeDatasThatNeedProcessingCount >= $configurationDto->getLimit()) {
+                break;
+            }
+
+            $currentFocusKeywordPropertyValue = $nodeData->hasProperty('focusKeyword') ? $nodeData->getProperty('focusKeyword') : null;
+            if (!self::matchFocusKeywordProperty($currentFocusKeywordPropertyValue, $configurationDto)) {
+                continue;
+            }
+
+            $nodeDatasThatNeedProcessing[] = $nodeData;
+            $nodeDatasThatNeedProcessingCount++;
+        }
+
+        $nodeDatasThatNeedProcessing = $this->reduceNodeVariantsByWorkspaces($nodeDatasThatNeedProcessing, $workspaceChain);
```

Comment:
Also du iterierst als erstes selber drüber und checkst "matchFocusKeywordProperty" um dann den reduce zu machen.
Wenn jetzt in Live die Property leer is, und in meinem Workspace bereits gesetzt, würd ich dann nicht nur das aus Live kriegen?
Ich glaub du darfst erst nach dem reduce den iterate-loop machen